lib/libutee/tee_api.c

```diff
@@ -245,6 +248,136 @@ TEE_Result TEE_InvokeTACommand(TEE_TASessionHandle session,
 	return res;
 }
 
+#ifdef CFG_OCALL
+TEE_Result TEE_InvokeCACommand(uint32_t cancellationRequestTimeout,
+			       uint32_t commandID, uint32_t paramTypes,
+			       TEE_Param params[TEE_NUM_PARAMS],
+			       uint32_t *returnOrigin)
+{
+	const TEE_UUID uuid = PTA_SYSTEM_UUID;
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t ret_origin = TEE_ORIGIN_TEE;
+	struct utee_params pta_up = { };
+	struct utee_params ocall_up = { };
+	void *buffer = NULL;
+	size_t size = 0;
+	size_t n = 0;
+
+	/* Open session with the System PTA, if necessary */
+	if (!__tee_api_system_session) {
```

Comment:
Very true. I modified `invoke_system_pta` to use this global variable instead.

After pushing I had another idea to clean this up, a few more changes incoming...

Reply:
No, the idea I had is uglier because it requires defining `tee_ocall` with the exact same parameters as `TEE_InvokeCACommand` in the exact same header file to avoid exposing `struct utee_params` via `tee_internal_api_extensions.h`.

Reply:
> Very true. I modified `invoke_system_pta` to use this global variable instead.

OK, looks reasonable.